tests/testsuite/features2.rs

```diff
@@ -2472,3 +2472,85 @@ fn all_features_merges_with_features() {
         .with_stdout("it works")
         .run();
 }
+
+#[cargo_test]
+fn dep_with_optional_host_deps_activated() {
+    // To prevent regression like rust-lang/cargo#11330
+    Package::new("serde_derive", "0.1.0")
+        .feature("derive", &["dep:serde_derive"])
+        .file("src/lib.rs", "")
+        .publish();
+
+    Package::new("serde", "0.1.0")
+        .feature("derive", &["dep:serde_derive"])
+        .dep("serde_derive", "0.1.0")
+        .file("src/lib.rs", "")
+        .publish();
```

Comment:
These lines seem unnecessary, since the test is only using path dependencies.

Or, alternatively, you can remove the path dependencies and use registry dependencies instead.